docs/spec.md

```diff
@@ -538,6 +561,20 @@ All integer types explicitly specified (uint32, uint64, etc.) in this section ar
 | _encoding length_ | char[]  | encoding, same encoding that was used for the request |
 | remaining bytes   | uint8[] | response payload                                      |
 
+### Fetch Asset Response
+
+- Response to a [fetch asset](#fetch-asset) request
+- Only supported if the server previously declared the `assets` [capability](#server-info).
+
+| Bytes                  | Type    | Description                                                                                                             |
+| ---------------------- | ------- | ----------------------------------------------------------------------------------------------------------------------- |
+| 1                      | opcode  | 0x04                                                                                                                    |
+| 4                      | uint32  | request id, as given in the corresponding [request](#fetch-asset)                                                       |
+| 1                      | uint8   | status enum, `0` for success and `1` for error. Values `>1` are reserved for future use                                 |
+| 4                      | uint32  | error message length                                                                                                    |
```

Comment:
not sure if worth it - but 64k ought to be enough for any error message 🤣

Reply:
I suppose uint32 is what we use as the length prefix for strings consistently so this seems 👍